src/common/reorder.cpp

```diff
@@ -70,16 +71,23 @@ status_t reorder_primitive_desc_create(std::shared_ptr<primitive_desc_t> &pd,
                     s_ek != d_ek, utils::one_of(engine_kind::cpu, s_ek, d_ek));
     if (!args_ok) return invalid_arguments;
 
-    auto s_mdw = memory_desc_wrapper(*src_md);
     auto d_mdw = memory_desc_wrapper(*dst_md);
 
-    if (!s_mdw.consistent_with(d_mdw)) return invalid_arguments;
+    // If src and dst are not already consistent, try to reshape them
+    memory_desc_t reshaped_src_md = *src_md;
+    auto reshaped_s_mdw = memory_desc_wrapper(&reshaped_src_md);
```

Comment:
This change applies to the whole library and changes API behavior.
If you want to have such feature we will need an RFC to discuss available options.

As an alternative option you could uncollapse weights md internally before returning to a user (or keep both collapsed and uncollapsed versions for internal use and API calls). This way reorders will work as expected and the change will be isolated to this specific inner product implementation.

Reply:
That's a fair point, I'll have a think and create an RFC if it still makes sense.

Would you be able to expand on what you mean by this

> As an alternative option you could uncollapse weights md internally in the implementation before returning to a user 

I think we need to pass the collapsed memory descriptor back to the user so that they can do the right reorder. The collapsed reorder is a different to the non-collapsed one. It's also beneficial, for example If we have a tensor with H=2, W=2 and IC=3 and we need to pad to 4, then
- 2x2x3 -pad-> 2x2x4 -collapse-> 16
- 2x2x3 -collapse-> 12 -pad-> 12